tools/diff-processor/breaking_changes/resource_diff_test.go

```diff
@@ -88,3 +107,112 @@ var resourceSchemaRule_RemovingAField_TestCases = []resourceSchemaTestCase{
 		expectedFields: []string{"field-a", "field-b"},
 	},
 }
+
+var resourceSchemaRule_AddingExactlyOneOf_TestCases = []resourceSchemaTestCase{
```

Comment:
We need a few more test cases:

1. [ ] Switching a field from required to ExactlyOneOf with just that field: not breaking
2. [ ] Switching a field from ExactlyOneOf with just that field to required: not breaking
3. [ ] Adding ExactlyOneOf to an existing required field (removing required) and also one or more newly-introduced fields: not breaking
4. [ ] Adding ExactlyOneOf to new fields that are all within a newly-added, optional ancestor: not breaking
5. [ ] Removing some fields from an ExactlyOneOf: not breaking
6. [ ] Removing an entire ExactlyOneOf: not breaking

Reply:
In the case of removing some fields from an ExactlyOneOf: wouldn't this be breaking?

For example, if we have exactly one of fields a b and c, and a user's config specifies field b, changing the exactly one of to a and c would break the user's config.

Reply:
you're correct, good catch! I've updated the list above.